screens/PrepareFace.tsx

```diff
@@ -1,85 +1,90 @@
-import { useNavigation } from '@react-navigation/native';
-import { Camera } from 'expo-camera'
-import React, { FunctionComponent, useEffect, useState } from 'react';
-import { StyleSheet, View } from 'react-native';
-import { Button, Text } from '../components/Themed';
-import { Ionicons }  from '@expo/vector-icons';
-
+import { useNavigation } from "@react-navigation/native";
+import { Camera } from "expo-camera";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { StyleSheet, View } from "react-native";
+import { Button, Text } from "../components/Themed";
+import { Ionicons } from "@expo/vector-icons";
 const PrepareFace: FunctionComponent = () => {
-    const navigation = useNavigation();
-    const [seeingFace, setSeeingFace] = useState(true);
-    const [permissions, setHasPermission] = useState(false);
-    useEffect(() => {
-        Camera.requestPermissionsAsync()
-            .then(response => {
-                setHasPermission(response.status === 'granted');
-            });
-       
-    }, []);
-    if (!permissions) {
-        return (<Text>Please give permissions</Text>)
-    }
-    return (
-        <View style={Styles.container}>
-            <Camera style={Styles.camera}
-                type={Camera.Constants.Type.front}
-                onFacesDetected={(face) => setSeeingFace(face.faces.length > 0)}
-                flashMode={Camera.Constants.FlashMode.auto}/> 
-            <View style={Styles.bottomRow}>
-                <Ionicons.Button style={Styles.button} name={seeingFace ? 'arrow-forward-circle' : 'hand-left'} color={!seeingFace ? 'red' : 'black'} onPress={() => {
-                    //@ts-ignore
-                    navigation.replace('Camera');
-                }} disabled={!seeingFace}>
-                    {seeingFace ? 'Begin Recording' : 'Please Show Your Face to Begin Recording'}
-                </Ionicons.Button>
-            </View>
-        </View>
-        
-    )
-}
+  const navigation = useNavigation();
+  const [seeingFace, setSeeingFace] = useState(true);
+  const [permissions, setHasPermission] = useState(false);
+  useEffect(() => {
+    Camera.requestPermissionsAsync().then((response) => {
+      setHasPermission(response.status === "granted");
+    });
+  }, []);
+  if (!permissions) {
+    return <Text>Please give permissions</Text>;
+  }
+  return (
+    <View style={Styles.container}>
+      <Camera
+        style={Styles.camera}
+        type={Camera.Constants.Type.front}
+        onFacesDetected={(face) => setSeeingFace(face.faces.length > 0)}
+        flashMode={Camera.Constants.FlashMode.auto}
+      />
+      <View style={Styles.bottomRow}>
+        <Ionicons.Button
+          style={Styles.button}
+          name={seeingFace ? "arrow-forward-circle" : "hand-left"}
+          color={!seeingFace ? "red" : "black"}
+          onPress={() => {
+            //@ts-ignore
+            navigation.replace("Camera");
```

Comment:
im sure theres a reason for the ignore, but why is it needed?

Reply:
"Navigate object on stack navigator has replace, but when you use useNavigation hook, it doesn't know what navigator you have. thus, it doesn't know if the replace function exists in TS statically, _but our big brains know_." - Jack

Reply:
LMAO sounds good